docs/ContinuousIntegration.md

```diff
@@ -0,0 +1,121 @@
+are several other workflows primarily for productization and packaging specific testing.
+
+### Product Testing
+
+Each product workflow has several jobs. The jobs typically matrix across all platforms the product
+supports.
+
+#### `pod lib lint` Tests
+
+Use CocoaPods to build, unit test, and run the Xcode Analyzer on the product.
+
+#### SPM Tests
+
+Use Swift Package Manager to build and usually unit test the product.
+
+#### QuickStart Tests
+
+Build and test the product's QuickStart.
+
+#### Integration Tests
+
+Run integration tests for the product. These either use a GitHub secret to access a Firebase project
+or use the Firebase emulator.
+
+#### Cron Tests
+
+Tests that only run in the nightly cron run. This is typically the `--static-framework` CocoaPods
+configuration that rarely breaks if dynamic frameworks are working.
+
+### Style Tests
+[check.yml](https://github.com/firebase/firebase-ios-sdk/tree/master/.github/workflows/check.yml)
+
+Runs several coding style tests. Details to come in an upcoming document about the scripts
+directory.
+
+### SwiftPM Tests
+[spm.yml](https://github.com/firebase/firebase-ios-sdk/tree/master/.github/workflows/spm.yml)
+
+Build and run Firebase-wide tests with Swift Package Manager.
+
+### Zip Distribution Testing
+[zip.yml](https://github.com/firebase/firebase-ios-sdk/tree/master/.github/workflows/zip.yml)
+
+Builds the zip distribution both from the tip of `master` and the current staged release distribution.
+The resulting distribution is then used to build and test several Firebase QuickStarts.
+
+### Release testing
+[scripts/create_spec_repo](https://github.com/firebase/firebase-ios-sdk/tree/master//scripts/create_spec_repo)
+
+Release testing is to build up a testing podspecs (CocoaPods podspecs) candidate and test building
+up a quickstart with CocoaPods. If this candidate is successfully built, that means tests in
+podspecs, e.g. abtesting test spec, passed and it is a positive signal to build up a release
+candidate.
+
+Currently we have two workflows running nightly to test podspecs:
+
+#### Release workflow
+[release.yml](https://github.com/firebase/firebase-ios-sdk/tree/master/.github/workflows/release.yml)
+
+The release workflow is to test podspecs corresponding to the latest release tag in the repo, and
+create a CocoaPods spec testing repo. Podspecs in this testing repo
+will have tags `Cocoapods-X.Y.Z`. This is to mimic a real released candidate.
+
+#### Prerelease workflow
+[prerelease.yml](https://github.com/firebase/firebase-ios-sdk/tree/master/.github/workflows/prerelease.yml)
+
+The prerelease workflow is to test podspecs on the `master` branch, and create a testing repo. This is
+to make sure podspecs are releasable, which means podspecs in the head can pass all tests and build
+up a candidate. The next step is to enable presubmit for PRs of podspecs changes on the prerelease
+testing repo. This will accelerate the tests of podspecs, which are currently run nightly.
```

Comment:
I think this has been done. 
https://github.com/firebase/firebase-ios-sdk/blob/master/.github/workflows/spectesting.yml
The sepctesting workflow can detect updated files and trigger corresponding podspec testing.

Reply:
Added a SpecTesting workflow section. PTAL